ballerina-shell/modules/shell-core/src/main/java/io/ballerina/shell/parser/trials/ModuleMemberTrial.java

```diff
@@ -60,4 +66,18 @@ public Node parseSource(String source) throws ParserTrialFailedException {
         }
         return dclnNode;
     }
+
+    private void validateModuleDeclaration(ModuleMemberDeclarationNode declarationNode) throws InvalidMethodException {
+        if (declarationNode instanceof FunctionDefinitionNode) {
+            String functionName = ((FunctionDefinitionNode) declarationNode).functionName().text();
+            if (RESTRICTED_FUNCTION_NAMES.contains(functionName)) {
+                String message = "Function name " + "'" + functionName + "'" + " not allowed in Ballerina Shell.\n";
+                throw new InvalidMethodException(message);
+            }
+            if (functionName.startsWith("__")) {
```

Comment:
What't the reason for this restriction? 🤔

Reply:
Current shell `.bal` file template uses function names starting from `__`. Because of that reason it is required to avoid functions starting from `__`

Reply:
Shall we create an issue for that? That's something we could look at improving, later on.

Reply:
https://github.com/ballerina-platform/ballerina-lang/issues/32493